components/dashboard/src/projects/Prebuilds.tsx

```diff
@@ -228,4 +229,63 @@ export function prebuildStatusIcon(status: PrebuiltWorkspaceState | undefined) {
         default:
             break;
     }
+}
+
+export function PrebuildInstanceStatus(props: { prebuildInstance?: WorkspaceInstance, isDark?: boolean }) {
+    let status = <></>;
+    let details = <></>;
+    switch (props.prebuildInstance?.status.phase) {
+        case undefined: // Fall through
+        case 'unknown':
+            status = <div className="flex space-x-1 items-center text-yellow-600">
+                <img className="h-4 w-4" src={StatusPaused} />
+                <span>PENDING</span>
+                </div>;
+            details = <div className="flex space-x-1 items-center text-gray-400">
+                <img className="h-4 w-4 animate-spin" src={props.isDark ? SpinnerDark : Spinner} />
+                <span>Prebuild in progress ...</span>
+                </div>;
+            break;
+        case 'preparing': // Fall through
+        case 'pending': // Fall through
+        case 'creating': // Fall through
+        case 'initializing': // Fall  through
+        case 'running': // Fall through
+        case 'interrupted':
+            status = <div className="flex space-x-1 items-center text-blue-600">
+                <img className="h-4 w-4" src={StatusRunning} />
+                <span>RUNNING</span>
+                </div>;
+            details = <div className="flex space-x-1 items-center text-gray-400">
+                <img className="h-4 w-4 animate-spin" src={props.isDark ? SpinnerDark : Spinner} />
+                <span>Prebuild in progress ...</span>
+                </div>;
+            break;
+        case 'stopping': // Fall through
+        case 'stopped':
+            status = <div className="flex space-x-1 items-center text-green-600">
```

Comment:
**issue**: Noticed that the status on the header section does not automatically get updated but you need to reload to the get new status. This could lead to having a discrepancy between these two places where status is surfaced.

| Example A |  Example B | 
|-|-|
| <img width="505" alt="Frame 218" src="https://user-images.githubusercontent.com/120486/131985193-e5d2a1a0-5e59-4609-9950-3930bc811ad5.png"> | <img width="505" alt="Frame 2182" src="https://user-images.githubusercontent.com/120486/131985199-b9dc4809-2009-48ca-a61c-3df90f0489c5.png"> |

Reply:
Agreed: The top status is based on the "prebuild" status, while the bottom one is based on the "prebuild workspace instance" status. Normally they should be in sync, but discrepancies can happen during incidents or because of bugs.